sqlite3.pc Includes Unnecessary -ldl Flag on Windows, Breaking Builds

### Description / Steps to reproduce the issue

The mingw-w64-x86_64-sqlite3 package (version 3.50.3) includes -ldl in the Libs.private field of sqlite3.pc, causing linker errors (ld: cannot find -ldl) in MSYS2 MINGW64 when using pkg-config --static --libs sqlite3. The libdl library is not needed on Windows, as dynamic linking uses the Windows API.




### Expected behavior


pkg-config --static --libs sqlite3

Output: -LC:/msys64/mingw64/lib -lsqlite3 -lz

### Actual behavior

pkg-config --static --libs sqlite3

Output: -LC:/msys64/mingw64/lib -lsqlite3 -lz -ldl

### Verification

- [x] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [x] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANGARM64

### Are you willing to submit a PR?

_No response_

--- a/mingw-w64-sqlite3/0005-remove-dl-flags.patch
+++ b/mingw-w64-sqlite3/0005-remove-dl-flags.patch
@@ -1,0 +1,9 @@
+--- sqlite-src-3500300/sqlite3.pc.in
++++ sqlite-src-3500300/sqlite3.pc.in
+@@ -9,5 +9,5 @@ Name: SQLite
+ Description: SQL database engine
+ Version: @PACKAGE_VERSION@
+ Libs: -L${libdir} -lsqlite3
+-Libs.private: @LDFLAGS_MATH@ @LDFLAGS_ZLIB@ @LDFLAGS_DLOPEN@ @LDFLAGS_PTHREAD@ @LDFLAGS_ICU@
++Libs.private: @LDFLAGS_MATH@ @LDFLAGS_ZLIB@ @LDFLAGS_PTHREAD@ @LDFLAGS_ICU@
+ Cflags: -I${includedir}

--- a/mingw-w64-sqlite3/PKGBUILD
+++ b/mingw-w64-sqlite3/PKGBUILD
@@ -10,7 +10,7 @@ _sqlite_year=2025
 _amalgamationver=3500300
 _docver=${_amalgamationver}
 pkgver=3.50.3
-pkgrel=1
+pkgrel=2
 pkgdesc="A C library that implements an SQL database engine (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clangarm64')
@@ -33,6 +33,7 @@ source=(https://www.sqlite.org/${_sqlite_year}/sqlite-src-${_amalgamationver}.zi
         0002-fix-building-sqldiff.patch
         0003-do-not-install-tclsqlite3-with-make.patch
         0004-dont-misuse-tcl-extern.patch
+        0005-remove-dl-flags.patch
         Makefile.ext.in
         README.md.in
         LICENSE)
@@ -42,6 +43,7 @@ sha256sums=('119862654b36e252ac5f8add2b3d41ba03f4f387b48eb024956c36ea91012d3f'
             'cf5c47c30e97f5493d2fad730a9bfcd33d20a0a052571e8e35b88f388d368724'
             'a500428a3434075932de84247b1405f3d73e7ed0a8b664eb20a736bd02d6ed29'
             '8dd2377afaad17fa85049be90bfc2bd757350343b3f64ff9bfbbcf3736593a34'
+            '1f6c78b81b1cf426bb582973a8a18557f14c7e96de3e0080f209c9c3f2a3b1a1'
             '84b64569978dde5b63799099b0fbb3336943ef877170beb84483020c67e25b3c'
             '5ca42f1f92abfb61bacc9ff60f5836cc56e2ce2af52264f918cb06c3d566d562'
             '0b76663a90e034f3d7f2af5bfada4cedec5ebc275361899eccc5c18e6f01ff1f')
@@ -54,6 +56,7 @@ prepare() {
   patch -p1 -i "${srcdir}/0001-sqlite-pcachetrace-include-sqlite3.patch"
   patch -p1 -i "${srcdir}/0002-fix-building-sqldiff.patch"
   patch -p1 -i "${srcdir}/0004-dont-misuse-tcl-extern.patch"
+  patch -p1 -i "${srcdir}/0005-remove-dl-flags.patch"
 }
 
 build() {

